src/main/java/domain/Lotto.java

```diff
@@ -10,48 +11,57 @@
 
 public class Lotto {
 
-    public static final int NUMBERS_SIZE = 6;
+    private static final int NUMBERS_SIZE = 6;
 
     private final Set<Number> numbers;
 
-    public Lotto(Set<Number> numbers) {
-        validateNumberSize(numbers);
+    private Lotto(Set<Number> numbers) {
         this.numbers = numbers;
     }
 
-    public Lotto(List<Number> numbers) {
+    public static Lotto createWinningLotto(List<Integer> number) {
+        List<Number> numbers = number.stream().map(Number::new).toList();
         validateNumberSize(numbers);
         validateNonDuplicatedNumbers(numbers);
-        this.numbers = new HashSet<>(numbers);
+        return new Lotto(new HashSet<>(numbers));
     }
 
-    public boolean contains(Number number) {
-        return numbers.contains(number);
+    public static Lotto createRandomLotto(LottoPickStrategy lottoPickStrategy) {
+        Set<Number> numbers;
+        do {
+            numbers = new HashSet<>(selectNumbers(lottoPickStrategy));
+        } while (numbers.size() != NUMBERS_SIZE);
+        return new Lotto(numbers);
+    }
+
+    private static List<Number> selectNumbers(LottoPickStrategy lottoPickStrategy) {
+        List<Integer> numbers = lottoPickStrategy.pickNumbers(NUMBERS_SIZE);
+        return numbers.stream().map(Number::new).toList();
```

Comment:
```suggestion
    public static Lotto createRandomLotto(LottoPickStrategy lottoPickStrategy) {
        Set<Integer> numbers = lottoPickStrategy.pickNumbers(NUMBERS_SIZE);
        return new Lotto(lottoPickStrategy.pickNumbers(NUMBERS_SIZE));
    }
```
`LottoPickStrategy`에서 나온 값을 사용하는 곳이 여기에서만 사용되고 `Lotto`는 `Set`만을 사용하고 있으니 `LottoPickStrategy`의 반환값 자체를 `Set`으로 두면 변환 로직이 필요없을 것 같아요.

그리고 파라메터로 6개를 준만큼 나오는 것을 기대할테니 while문이 의미가 없어보여요.